fog/view/server/tests/router_smoke_tests.rs

```diff
@@ -93,28 +218,210 @@ async fn test_router_integration(test_environment: &mut RouterTestEnvironment, l
+    assert_eq!(result.next_start_from_user_event_id, 80);
+    assert_eq!(result.rng_records.len(), 0);
+    assert_eq!(result.tx_out_search_results.len(), 1);
+    assert_eq!(
+        TxOutSearchResultCode::try_from(result.tx_out_search_results[0].result_code).unwrap(),
+        TxOutSearchResultCode::BadSearchKey
+    );
     assert_eq!(result.missed_block_ranges.len(), 0);
-    assert_eq!(result.last_known_block_count, 5);
+    assert_eq!(result.last_known_block_count, 6);
+
+    let result = router_client
+        .query(4, 0, vec![vec![1u8; 16], vec![2u8; 16], vec![3u8; 16]])
+        .await;
+    assert!(result.is_ok());
+    let result = result.unwrap();
+    assert_eq!(result.highest_processed_block_count, 6);
+    assert_eq!(result.next_start_from_user_event_id, 4);
+    assert_eq!(result.rng_records.len(), 0);
+    assert_eq!(result.tx_out_search_results.len(), 3);
+    {
+        let sort_txs = interpret_tx_out_results(result.tx_out_search_results.clone());
+        assert_eq!(sort_txs[0].search_key, vec![1u8; 16]);
+        assert_eq!(sort_txs[0].result_code, 1);
+        assert_eq!(sort_txs[0].ciphertext, vec![1u8; 232]);
+
+        assert_eq!(sort_txs[1].search_key, vec![2u8; 16]);
+        assert_eq!(sort_txs[1].result_code, 1);
+        assert_eq!(sort_txs[1].ciphertext, vec![2u8; 232]);
+
+        assert_eq!(sort_txs[2].search_key, vec![3u8; 16]);
+        assert_eq!(sort_txs[2].result_code, 1);
+        assert_eq!(sort_txs[2].ciphertext, vec![3u8; 232]);
+    }
+    assert_eq!(result.missed_block_ranges.len(), 0); // no range reported since we started at event id 4
+    assert_eq!(result.last_known_block_count, 6);
+
+    let result = router_client
+        .query(4, 0, vec![vec![5u8; 16], vec![8u8; 16], vec![200u8; 16]])
+        .await;
+    assert!(result.is_ok());
+    let result = result.unwrap();
+    assert_eq!(result.highest_processed_block_count, 6);
+    assert_eq!(result.next_start_from_user_event_id, 4);
+    assert_eq!(result.rng_records.len(), 0);
+    assert_eq!(result.tx_out_search_results.len(), 3);
+    {
+        let sort_txs = interpret_tx_out_results(result.tx_out_search_results.clone());
+        assert_eq!(sort_txs[0].search_key, vec![5u8; 16]);
+        assert_eq!(sort_txs[0].result_code, 1);
+        assert_eq!(sort_txs[0].ciphertext, vec![5u8; 232]);
+
+        assert_eq!(sort_txs[1].search_key, vec![8u8; 16]);
+        assert_eq!(sort_txs[1].result_code, 1);
+        assert_eq!(sort_txs[1].ciphertext, vec![8u8; 232]);
+
+        assert_eq!(sort_txs[2].search_key, vec![200u8; 16]);
+        assert_eq!(sort_txs[2].result_code, 2);
+        // FIGURE OUT WHAT TO DO HERE
+        assert_eq!(sort_txs[2].ciphertext, vec![0u8; 254]);
+    }
+
+    assert_eq!(result.missed_block_ranges.len(), 0); // no range reported since we started at event id 4
+    assert_eq!(result.last_known_block_count, 6);
+
+    let result = router_client.query(0, 0, vec![vec![200u8; 17]]).await;
```

Comment:
:question: What is this test doing over the one above that looked for `[200u8: 16]`?  Is it the extra key length?

Reply:
Yes, the extra length makes the key invalid so you get a different error code (invalid key instead of key not found).